### PR TITLE
fix: should recognise breaking changes

### LINKED ISSUE #507: 🐛 Bug: `BREAKING CHANGE:` not recognised
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/should-semantic-release/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`BREAKING CHANGE:` to be detected as a breaking change release

### Actual

```
Checking commit: BREAKING CHANGE: registry feature and change shortcuts
Found type undefined. Continuing.
```

See https://github.com/johnnyreilly/azdo-npm-auth/actions/runs/12215799512/job/34078200073#step:5:11

### Additional Info

https://github.com/johnnyreilly/azdo-npm-auth/pull/20

It looks that `feat!:` is also not recognised - same log messages:

https://github.com/johnnyreilly/azdo-npm-auth/pull/21

https://github.com/johnnyreilly/azdo-npm-auth/actions/runs/12215949788/job/34078528733#step:5:11

Looking at https://github.com/JoshuaKGoldberg/should-semantic-release/pull/140 it suggests this ought to work. But it doesn't.  Or I'm missing something

### PR DESCRIPTION
**Pull request by johnnyreilly**
## PR Checklist

- [x] Addresses an existing open issue: fixes #507
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/should-semantic-release/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/should-semantic-release/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This PR amends behaviour to recognise BREAKING CHANGE as meaningful

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
I like it, seems very reasonable to me! Just requesting a bit of testing to exercise the edge case in the breaking change?

File: src/getCommitMeaning.test.ts [hunk 1]
```diff
@@ -72,8 +72,12 @@ describe("getCommitMeaning", () => {
...
 		],
 		[
-			"BREAKING CHANGE (major): line starts with something like BREAKING CHANGE",
-			{ type: undefined },
+			"BREAKING CHANGE: line starts with something like BREAKING CHANGE",
+			"meaningful",
+		],
+		[
+			"BREAKING CHANGE (major): line starts with something like BREAKING CHANGE and has modifier",
+			"meaningful",
```
[the remark this answers is not in the document]
Comment (johnnyreilly) [reply src/getCommitMeaning.test.ts:80]: I think I've just added what you want!

[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply src/getCommitMeaning.test.ts:80]: And then some 🔥

**Review by JoshuaKGoldberg — APPROVED** [review]
Breakdancer spinning on their back in a circle of onlookers

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @johnnyreilly for code.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙
